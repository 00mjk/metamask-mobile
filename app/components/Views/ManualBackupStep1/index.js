import React, { PureComponent } from 'react';
import { Text, View, SafeAreaView, StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import { colors, fontStyles } from '../../../styles/common';
import StyledButton from '../../UI/StyledButton';
import OnboardingProgress from '../../UI/OnboardingProgress';
import { strings } from '../../../../locales/i18n';
import FeatherIcons from 'react-native-vector-icons/Feather';
import { BlurView } from '@react-native-community/blur';
import ActionView from '../../UI/ActionView';
import Device from '../../../util/Device';

const styles = StyleSheet.create({
	mainWrapper: {
		backgroundColor: colors.white,
		flex: 1
	},
	wrapper: {
		flex: 1,
		paddingHorizontal: 32
	},
	onBoardingWrapper: {
		paddingHorizontal: 20
	},
	action: {
		fontSize: 18,
		marginBottom: 16,
		color: colors.fontPrimary,
		justifyContent: 'center',
		textAlign: 'center',
		...fontStyles.bold
	},
	infoWrapper: {
		marginBottom: 16,
		justifyContent: 'center'
	},
	info: {
		fontSize: 14,
		color: colors.fontPrimary,
		textAlign: 'center',
		...fontStyles.normal,
		paddingHorizontal: 6
	},
	seedPhraseConcealer: {
		position: 'absolute',
		width: '100%',
		height: '100%',
		backgroundColor: colors.grey700,
		opacity: 0.7,
		alignItems: 'center',
		borderRadius: 8,
		paddingHorizontal: 24,
		paddingVertical: 45
	},
	blurView: {
		position: 'absolute',
		top: 0,
		left: 0,
		bottom: 0,
		right: 0,
		borderRadius: 8
	},
	icon: {
		width: 24,
		height: 24,
		color: colors.white,
		textAlign: 'center',
		marginBottom: 32
	},
	reveal: {
		fontSize: Device.isMediumDevice() ? 13 : 16,
		...fontStyles.extraBold,
		color: colors.white,
		lineHeight: 22,
		marginBottom: 8,
		textAlign: 'center'
	},
	watching: {
		fontSize: Device.isMediumDevice() ? 10 : 12,
		color: colors.white,
		lineHeight: 17,
		marginBottom: 32,
		textAlign: 'center'
	},
	viewButtonContainer: {
		width: 155,
		padding: 12
	},
	seedPhraseWrapper: {
		backgroundColor: colors.white,
		borderRadius: 8,
		flexDirection: 'row',
		borderColor: colors.grey100,
		borderWidth: 1,
		marginBottom: 64,
		height: 275
	},
	wordColumn: {
		flex: 1,
		alignItems: 'center',
		paddingHorizontal: Device.isMediumDevice() ? 18 : 24,
		paddingVertical: 18,
		justifyContent: 'space-between'
	},
	wordWrapper: {
		flexDirection: 'row'
	},
	word: {
		paddingHorizontal: 8,
		paddingVertical: 6,
		fontSize: 14,
		color: colors.fontPrimary,
		backgroundColor: colors.white,
		borderColor: colors.blue,
		borderWidth: 1,
		borderRadius: 13,
		textAlign: 'center',
		lineHeight: 14,
		flex: 1
	}
});

/**
 * View that's shown during the second step of
 * the backup seed phrase flow
 */
export default class ManualBackupStep1 extends PureComponent {
	static propTypes = {
		/**
		/* navigation object required to push and pop other views
		*/
		navigation: PropTypes.object
	};
	constructor(props) {
		super(props);
		this.words = props.navigation.getParam('words');
	}
	steps = [
		strings('manual_backup.progressOne'),
		strings('manual_backup.progressTwo'),
		strings('manual_backup.progressThree')
	];

	state = {
		seedPhraseHidden: true,
		currentStep: 2
	};

	goNext = () => {
		this.props.navigation.navigate('ManualBackupStep2', { words: this.words, steps: this.steps });
	};

	revealSeedPhrase = () => this.setState({ seedPhraseHidden: false });

	renderSeedPhraseConcealer = () => (
		<React.Fragment>
			<BlurView blurType="light" blurAmount={5} style={styles.blurView} />
			<View style={styles.seedPhraseConcealer}>
				<FeatherIcons name="eye-off" size={24} style={styles.icon} />
				<Text style={styles.reveal}>{strings('manual_backup_step_1.reveal')}</Text>
				<Text style={styles.watching}>{strings('manual_backup_step_1.watching')}</Text>
				<View style={styles.viewButtonWrapper}>
					<StyledButton
						type={'view'}
						onPress={this.revealSeedPhrase}
						testID={'view-button'}
						containerStyle={styles.viewButtonContainer}
					>
						{strings('manual_backup_step_1.view')}
					</StyledButton>
				</View>
			</View>
		</React.Fragment>
	);

	render() {
		return (
			<SafeAreaView style={styles.mainWrapper}>
				<View style={styles.onBoardingWrapper}>
					<OnboardingProgress currentStep={this.state.currentStep} steps={this.steps} />
				</View>
				<ActionView
					confirmTestID={'manual-backup-step-1-continue-button'}
					confirmText={strings('manual_backup_step_1.continue')}
					onConfirmPress={this.goNext}
					confirmDisabled={this.state.seedPhraseHidden}
					showCancelButton={false}
					confirmButtonMode={'confirm'}
				>
					<View style={styles.wrapper} testID={'manual_backup_step_1-screen'}>
						<Text style={styles.action}>{strings('manual_backup_step_1.action')}</Text>
						<View style={styles.infoWrapper}>
							<Text style={styles.info}>{strings('manual_backup_step_1.info')}</Text>
						</View>
						<View style={styles.seedPhraseWrapper}>
							<View style={styles.wordColumn}>
								{this.words.slice(0, 6).map((word, i) => (
									<View key={`word_${i}`} style={styles.wordWrapper}>
										<Text style={styles.word}>{`${i + 1}. ${word}`}</Text>
									</View>
								))}
							</View>
							<View style={styles.wordColumn}>
								{this.words.slice(-6).map((word, i) => (
									<View key={`word_${i}`} style={styles.wordWrapper}>
										<Text style={styles.word}>{`${i + 7}. ${word}`}</Text>
									</View>
								))}
							</View>
							{this.state.seedPhraseHidden && this.renderSeedPhraseConcealer()}
						</View>
					</View>
				</ActionView>
			</SafeAreaView>
		);
	}
}
