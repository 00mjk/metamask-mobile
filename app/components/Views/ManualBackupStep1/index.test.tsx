import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ManualBackupStep1 from './';

const mockStore = configureMockStore();
const initialState = {
	user: { appTheme: 'light' },
};
const store = mockStore(initialState);

describe('ManualBackupStep1', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Provider store={store}>
				<ManualBackupStep1
					route={{
						params: {
							words: [
								'abstract',
								'accident',
								'acoustic',
								'announce',
								'artefact',
								'attitude',
								'bachelor',
								'broccoli',
								'business',
								'category',
								'champion',
								'cinnamon',
							],
						},
					}}
				/>
			</Provider>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
