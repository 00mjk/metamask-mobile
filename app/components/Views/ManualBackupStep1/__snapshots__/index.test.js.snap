// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManualBackupStep1 should render correctly 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingHorizontal": 20,
      }
    }
  >
    <OnboardingProgress
      currentStep={2}
      steps={
        Array [
          "Create Password",
          "Secure wallet",
          "Confirm seed phrase",
        ]
      }
    />
  </View>
  <ActionView
    cancelTestID=""
    cancelText="Cancel"
    confirmButtonMode="confirm"
    confirmDisabled={true}
    confirmTestID="manual-backup-step-1-continue-button"
    confirmText="Continue"
    confirmed={false}
    onConfirmPress={[Function]}
    showCancelButton={false}
    showConfirmButton={true}
  >
    <View
      style={
        Object {
          "flex": 1,
          "paddingHorizontal": 32,
        }
      }
      testID="manual_backup_step_1-screen"
    >
      <Text
        style={
          Object {
            "color": "#000000",
            "fontFamily": "CircularStd-Medium",
            "fontSize": 18,
            "fontWeight": "600",
            "justifyContent": "center",
            "marginBottom": 16,
            "textAlign": "center",
          }
        }
      >
        Write down your seed phrase
      </Text>
      <View
        style={
          Object {
            "justifyContent": "center",
            "marginBottom": 16,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "CircularStd-Medium",
              "fontSize": 14,
              "fontWeight": "400",
              "paddingHorizontal": 6,
              "textAlign": "center",
            }
          }
        >
          This is your seed phrase. Write it down on a paper and keep it in a safe place. You'll be asked to re-enter this phrase (in order) on the next step.
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderColor": "#d6d9dc",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "marginBottom": 64,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "paddingBottom": 4,
              "paddingLeft": 27,
              "paddingTop": 18,
            }
          }
        >
          <Text
            key="word_0"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            1. abstract
          </Text>
          <Text
            key="word_1"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            2. accident
          </Text>
          <Text
            key="word_2"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            3. acoustic
          </Text>
          <Text
            key="word_3"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            4. announce
          </Text>
          <Text
            key="word_4"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            5. artefact
          </Text>
          <Text
            key="word_5"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            6. attitude
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "flex": 1,
              "paddingBottom": 4,
              "paddingRight": 27,
              "paddingTop": 18,
            }
          }
        >
          <Text
            key="word_0"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            7. bachelor
          </Text>
          <Text
            key="word_1"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            8. broccoli
          </Text>
          <Text
            key="word_2"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            9. business
          </Text>
          <Text
            key="word_3"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            10. category
          </Text>
          <Text
            key="word_4"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            11. champion
          </Text>
          <Text
            key="word_5"
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "borderColor": "#037dd6",
                "borderRadius": 13,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "lineHeight": 14,
                "marginBottom": 14,
                "paddingHorizontal": 8,
                "paddingVertical": 6,
                "textAlign": "center",
                "width": 95,
              }
            }
          >
            12. cinnamon
          </Text>
        </View>
        <BlurView
          blurAmount={5}
          blurType="light"
          style={
            Object {
              "borderRadius": 8,
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3C3F42",
              "borderRadius": 8,
              "height": "100%",
              "opacity": 0.7,
              "paddingVertical": 45,
              "position": "absolute",
              "width": "100%",
            }
          }
        >
          <Icon
            allowFontScaling={false}
            name="eye-off"
            size={24}
            style={
              Object {
                "color": "#FFFFFF",
                "height": 24,
                "marginBottom": 32,
                "textAlign": "center",
                "width": 24,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "CircularStd-Medium",
                "fontSize": 16,
                "fontWeight": "bold",
                "lineHeight": 22,
                "marginBottom": 8,
                "marginHorizontal": 38,
                "textAlign": "center",
              }
            }
          >
            Tap to reveal your seed phrase
          </Text>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 12,
                "lineHeight": 17,
                "marginBottom": 32,
              }
            }
          >
            Make sure no one is watching your screen.
          </Text>
          <View>
            <StyledButton
              containerStyle={
                Object {
                  "padding": 12,
                  "width": 155,
                }
              }
              disabledContainerStyle={
                Object {
                  "opacity": 0.6,
                }
              }
              onPress={[Function]}
              styleDisabled={
                Object {
                  "opacity": 0.6,
                }
              }
              testID="view-button"
              type="view"
            >
              View
            </StyledButton>
          </View>
        </View>
      </View>
    </View>
  </ActionView>
</ForwardRef(SafeAreaView)>
`;
